### ISSUE TITLE
Does not support ! pegjs operator

### ISSUE THREAD

**Issue by Pyrolistical** [dundalek/GrammKit#9]
On your example site if you load the JavaScript grammar we have
```
ExpressionStatement
  = !("{" / FunctionToken) expression:Expression EOS {
      return {
        type:       "ExpressionStatement",
        expression: expression
      };
    }
```

and the diagram looks like
<img width="489" alt="screen shot 2017-09-10 at 8 16 34 pm" src="https://user-images.githubusercontent.com/463642/30257225-442dc142-9665-11e7-99d9-8ea79b334526.png">

As far as I understand, that is an invalid railroad diagram.

**Comment by dundalek** [comment]
Yeah, that's a negative lookahead. I represent it as a branch with two slashes before it (as you can see on the diagram). The idea is to indicate it is there, but you cannot go through it so it is not part of the parsed result.

Do you have a better way to represent it in mind?

**Comment by Pyrolistical** [comment]
Without words, I think this is more clear
<img width="597" alt="screen shot 2017-09-21 at 6 41 05 pm" src="https://user-images.githubusercontent.com/463642/30725868-795ee63a-9efc-11e7-9d0b-b12b1bcf4619.png">

**Comment by dundalek** [comment]
Hmm, I guess I slightly prefer the first arrangement, since it indicates that parser checks the lookahead first and then continues with the rest. But I like the idea of the cross symbol, I might try to use that since it is distinguishable from the start/end symbols.

**Comment by dundalek** [comment]
@Pyrolistical What do you think of this?

![image](https://user-images.githubusercontent.com/755611/30747773-86ba9d62-9fae-11e7-85bb-6d51b5901c21.png)

**Comment by Pyrolistical** [comment]
I don't see the value of having the X in the front nor having it connect back to the beginning of Expression. Yes this is describing more closely to what the parser is doing, but this is a railroad diagram. I feel my version is more in the spirit of a railroad diagram. Every line you follow is a valid syntax, except we are adding a concept of an invalid path with the X

**Comment by jakwings** [comment]
How about this?

(fig.1 all sub items in the lookahead branch keep using dashed border lines)
<img width="594" alt="fig.1" src="https://user-images.githubusercontent.com/23622455/33819607-4775859c-de86-11e7-8510-32ab830ece54.png">

(fig.2 only markers)
<img width="612" alt="fig.2" src="https://user-images.githubusercontent.com/23622455/33824540-b5f17568-de99-11e7-8f94-8c9a188329e5.png">

**Comment by iology** [comment]
Also we can use dashed lines for the borders.

<img width="245" src="https://user-images.githubusercontent.com/23622455/33816094-85febff0-de71-11e7-8654-3b74aa9ad310.png">

**Comment by jakwings** [comment]
Or like this:

(fig.3 use `(` and `)` to mark the start and end)

<img width="640" alt="fig.3" src="https://user-images.githubusercontent.com/23622455/33826705-da4ea834-dea0-11e7-874b-54fa5ffc7e7c.png">

**Comment by jakwings** [comment]
And this:

(fig.4 mark the area with dashed lines)
<img width="586" alt="fig 4" src="https://user-images.githubusercontent.com/23622455/33860174-752960bc-df12-11e7-84e6-8fe77f9ae2e3.png">

**Comment by dundalek** [comment]
Thanks for the suggestions @iology. I like the dashed variant in fig.1. However, it still feels a bit inconclusive for me to implement it.

**Comment by jakwings** [comment]
I like fig.4 the most, but I may need to modify the railroad-diagram package. (looking into it)

fig.1 and fig.2 do not work quite nice when there is no branching and when there are nested lookaheads. fig.3 is a bit verbose and we have to find the corresponding start&end markers.
